.github/workflows/CI.yml

```diff
@@ -58,12 +66,16 @@ jobs:
     runs-on: ${{ matrix.os }}
     steps:
       - name: "Checkout repository 🛎"
-        uses: actions/checkout@v4
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
```

Comment:
Is this pin a security thing (in case actions get malicious injection)? Or some other reason? I'm surprised to see it on `actions/checkout` since it's provided by the platform